pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/store/TemporaryCredentialsStore.scala

```diff
@@ -0,0 +1,21 @@
+package uk.ac.wellcome.platform.transformer.mets.store
+
+import com.amazonaws.services.s3.AmazonS3
+import uk.ac.wellcome.platform.transformer.mets.client.AssumeRoleClientProvider
+import uk.ac.wellcome.storage.ObjectLocation
+import uk.ac.wellcome.storage.store.s3.S3TypedStore
+import uk.ac.wellcome.storage.streaming.Codec
+
+class TemporaryCredentialsStore[T](
+  assumeRoleClientProvider: AssumeRoleClientProvider[AmazonS3])(
```

Comment:
Maybe this could just be a `ClientFactory[AmazonS3]` to abstract this away further? Would mean the knowledge of credentials being temporary were not exposed here, and within `MetsTransformerWorkerService`

Reply:
Oops misunderstood the type hierarchy, so I don't  think that is possible

Reply:
How would it abstract further though? if It only takes a `ClientFactory` it will need to instantiate a `AssumeRoleClientProvider` itself, but to do that it will need the `stsClient` and the `roleArn` as additional parameters. It feels like more stuff about assuming the role would be leaking into this class rather than being isolated in  `AssumeRoleClientProvider`.
But perhaps I'm not getting what you mean?

Reply:
Yup I misread it, ignore me